File: Java/src/Animal.java
abstract class Animal {
    String name;
    Animal(String name) {
        this.name = name;
    }
    String getName() {
        return this.name;
    }
}

class Dog extends Animal {
    Dog(String name) {
        super(name);
    }
}

class Cat extends Animal {
    Cat(String name) {
        super(name);
    }
}

class Application {
    // return the name of the given animal
    public static String getAnimalName(Animal animal) {
        return animal.getName();
    }
}

File: Java/kotlin/Animal.kt
package kotlin

abstract class Animal(val name: String) {
    fun getName(): String = name
}

class Dog(name: String) : Animal(name)

class Cat(name: String) : Animal(name)

object Application {
    // return the name of the given animal
    fun getAnimalName(animal: Animal): String {
        return animal.getName()
    }
}

